docs: Add missing kernel options to system requirements

yrsuthari opened the pull request:
# Add missing kernel options to system requirements

This PR adds the following kernel options to the system requirements documentation:

- `CONFIG_NETFILTER_XT_MATCH_COMMENT` for iptables-based masquerading and k3s compatibility
- `CONFIG_VXLAN` for tunneling between nodes (default mode)
- `CONFIG_FIB_RULES` for policy routing used by Cilium

These options were found to be necessary when running Cilium on a custom kernel, but were not documented in the system requirements.

## Changes
- Added `CONFIG_NETFILTER_XT_MATCH_COMMENT` to the "Requirements for Iptables-based Masquerading" section
- Added a new "Requirements for Tunneling and Routing" section
- Added `CONFIG_VXLAN` and `CONFIG_FIB_RULES` to the new section with explanations

```release-note
docs: Add missing kernel options to system requirements documentation to help users with custom kernels.
```

Fixes: #36189

yrsuthari committed:
docs: Add missing kernel options to system requirements

Signed-off-by: Yogi Suthari <yrsuthari@gmail.com>

YutaroHayakawa reviewed (changes requested):
Thanks! Could you add the one for GENEVE as well?

yrsuthari committed:
docs: Add CONFIG_GENEVE to required kernel options

Add CONFIG_GENEVE kernel configuration option to the system requirements documentation as it is needed for Geneve-based tunneling between nodes.

Signed-off-by: Yogi Suthari <yrsuthari@gmail.com>

YutaroHayakawa reviewed (changes requested):
The changes look good. Please don't use merge commit and rebase on top of the latest main instead.
